pkg/custompluginmonitor/custom_plugin_monitor.go

```diff
@@ -0,0 +1,172 @@
+/*
+Copyright 2017 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package custompluginmonitor
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"time"
+
+	cpmtypes "k8s.io/node-problem-detector/pkg/custompluginmonitor/types"
+	"k8s.io/node-problem-detector/pkg/types"
+
+	"github.com/golang/glog"
+	"k8s.io/node-problem-detector/pkg/custompluginmonitor/plugin"
+	"k8s.io/node-problem-detector/pkg/util/tomb"
+)
+
+type customPluginMonitor struct {
+	config     cpmtypes.CustomPluginConfig
+	conditions []types.Condition
+	plugin     *plugin.Plugin
+	resultChan <-chan cpmtypes.Result
+	statusChan chan *types.Status
+	tomb       *tomb.Tomb
+}
+
+// NewCustomPluginMonitorOrDie create a new customPluginMonitor, panic if error occurs.
+func NewCustomPluginMonitorOrDie(configPath string) types.Monitor {
+	l := &customPluginMonitor{
```

Comment:
nit: do you want to change `l` to `c` or something else?

Reply:
Will do. It is really difficult to recognize. :)